src/Components/Space.tsx

```diff
@@ -126,9 +126,11 @@ const Space = memo(({
   // add into movePawn click, if space is timer, pause timer!
   const movePawn = async () => {
     const docSnap = await getDoc(gameState.roomId);
+    
 
     if (docSnap.exists()) {
       const newRoomValue = {...docSnap.data()}
+      const playersArray = docSnap.data().players
```

Comment:
nitpicking, you may as well use the newRoomValue for playersArray, instead of docSnap.data() again